### PR TITLE
Add Google sheet capability (#10) fixes #9

### LINKED ISSUE #9: [FEAT] allow JSON generation from google sheet link / ID
**What would you like to be added**:

Currently, we require the developer to download CSV of translations and thereafter generate language JSONs. However, we can also implement a method where the user can pass in google sheet link / ID in place of the path to CSV and we would generate the language JSONs.

**Why is this needed**:

Eliminates a manual step and saves time.

---

A good place to start for the issue is in the `csv_to_vanilla-i18n` folder and look through the present scripts.

### PR DESCRIPTION
**Pull request by RandallWert**
# Pull Request

## This change creates the capability of reading a Google sheet, not just a local csv file.

- [√] Major (I think?)
- [ ] Minor
- [ ] Patch

### CODE REVIEW

**Review by thealphadollar — CHANGES_REQUESTED** [review]
I tested and it works perfectly. Thanks @RandallWert . I would suggest some sanitation changes.

- Please update the code in the existing python script instead of adding a new one.
- Have default option for worksheet set as "Sheet1". When the user passes only a single param, check if it's a file path. Otherwise assume it to be sheet ID.
- I don't think we should allow entry of URL since it may not be consistent. For instance, I believed a direct sheet link from the browser had to be put. I was wrong.
- Update the readme with details on how the script works. The existing README.md inside `csv_to_vanilla-i18n` already contains some instructions.

**Comment by RandallWert** [comment]
Hi, thanks for your feedback! I understand all of your points except the third one. I'm not sure exactly what I should change for that one. Should I just make a note in the readme, instructing the user not to enter an entire URL, but only the sheet ID (and possibly the worksheet name, if different from the default)?

**Comment by thealphadollar** [comment]
On the third point, I meant to remove the feature to add the URL as a specific URL pattern is required. Users may be more confused with that.

**Comment by RandallWert** [comment]
OK, I'm trying to prevent that confusion by putting the URL pattern in the code, rather than asking the user to enter it. The user only has to be concerned with the sheet ID (and possibly the worksheet name, if different than the default).

I did my best to make the changes you requested to the Python script and the readme. I just pushed another commit with those changes.

Thank you again for the opportunity to work on this, and please let me know if I should refine the code any further.

File: csv_to_vanilla-i18n/README.md [hunk 1]
```diff
@@ -10,7 +10,17 @@ For reference, please check `example.csv` and generated language JSON files in `
 
 NOTE: Python3 should be installed to use the script, follow instructions [here](https://realpython.com/installing-python/) to install it.
 
-To run the script, `python3 csv_to_vanilla_i18n.py PATH_TO_LANGUAGE_CSV`.
+To run the script, `python3 csv_to_vanilla_i18n.py [parameters]`.
+
+For parameters in the above line, you have several options:
+
+* You can specify the sheet ID and worksheet name of a Google Sheets worksheet, in that order, separated by a space. (Do NOT enter the entire URL to the Google Sheet.)
```
Comment (thealphadollar) [inline csv_to_vanilla-i18n/README.md]:
```suggestion
* You can specify the sheet ID and worksheet name of a Google Sheets worksheet, in that order, separated by a space. Eg. `python3 csv_to_vanilla_i18n.py 1KfPLetq4VUvEApiGtWEUuQUhAYCrocyLmyiHV2cl_ks Sheet1`
```

File: csv_to_vanilla-i18n/README.md [hunk 2]
```diff
@@ -10,7 +10,17 @@ For reference, please check `example.csv` and generated language JSON files in `
...
+
+* You can specify only the sheet ID and let the worksheet name default to ‘Sheet1’.
```
Comment (thealphadollar) [inline csv_to_vanilla-i18n/README.md]:
```suggestion
* You can specify only the sheet ID and let the worksheet name default to ‘Sheet1’. Eg. `python3 csv_to_vanilla_i18n.py 1KfPLetq4VUvEApiGtWEUuQUhAYCrocyLmyiHV2cl_ks`
```

File: csv_to_vanilla-i18n/README.md [hunk 3]
```diff
@@ -10,7 +10,17 @@ For reference, please check `example.csv` and generated language JSON files in `
...
+
+* You can specify only the path to a local language CSV file. Be sure to include the .csv extension.
```
Comment (thealphadollar) [inline csv_to_vanilla-i18n/README.md]:
```suggestion
* You can specify only the path to a local language CSV file. Be sure to include the .csv extension. Eg. `python3 csv_to_vanilla_i18n.py /path/to/csv`
```

File: csv_to_vanilla-i18n/README.md [hunk 4]
```diff
@@ -10,7 +10,17 @@ For reference, please check `example.csv` and generated language JSON files in `
...
+
+* You can omit the parameters completely. Then you will be prompted for the name of a local language CSV file. Be sure to include the .csv extension.
```
Comment (thealphadollar) [inline csv_to_vanilla-i18n/README.md]:
```suggestion
* You can omit the parameters completely. Then you will be prompted for the name of a local language CSV file. Be sure to include the .csv extension. Eg. `python3 csv_to_vanilla_i18n.py`
```

File: csv_to_vanilla-i18n/csv_or_sheet_to_vanilla_i18.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
```
Comment (thealphadollar) [inline csv_to_vanilla-i18n/csv_or_sheet_to_vanilla_i18.py]: Please remove this file as the changes have been made in the original py script.

**Review by thealphadollar — CHANGES_REQUESTED** [review]
LGTM. Thanks for adding exhaustive details to the README.

I've requested some more minor changes. Please push them and we can merge the changes.

**Comment by RandallWert** [comment]
OK, made those changes to the readme, and hopefully removed that unnecessary .py script successfully. (That's my first experience with that -- I'm still pretty new to making open source contributions.)

**Review by thealphadollar — APPROVED** [review]
LGTM. Your contribution is amazing considering it's your first time to open source. Looking forward to interacting more with you Randall.

**Comment by RandallWert** [comment]
Thank you so much! It was a pleasure and helped me learn a lot! I appreciate the opportunity to contribute, and I look forward to more collaboration with you.

**Pull request merged**